e2e-test/src/tests/upload.test.js

```diff
@@ -0,0 +1,44 @@
+import { vBtn } from 'jest-puppeteer-vuetify';
+import { CLIENT_URL } from '../util';
+
+
+describe('upload dataset', () => {
+  it('upload one dataset and verify it', async () => {
+    await Promise.all([
+      page.goto(CLIENT_URL),
+      page.waitForNavigation({ waitUntil: 'networkidle0' }),
+    ]);
+    await expect(page).toClickXPath(vBtn('Your Datasets'));
+    const fileInputElementHandle = await page.waitForXPath("//input[@type='file']");
+    await fileInputElementHandle.uploadFile('../e2e-test/data/Cancer chemo muscle (NMR).csv');
+    const fileName = await page.waitForXPath("//div[@class='v-list__tile__title'][contains(text(),'Cancer_chemo_muscle_NMR.csv ')]");
+    expect(fileName).not.toBeNull();
+    const datasetStatus = await page.waitForXPath("//div[@class='v-list__tile__title'][contains(text(),'Cancer_chemo_muscle_NMR.csv ')]//following::span[contains(text(),'Dataset processed with 1 validation failures')]");
+    expect(datasetStatus).not.toBeNull();
+  });
+  it('upload 3 dataset and verify it', async () => {
```

Comment:
This might be redundant and I can remove it if needed.

Reply:
I don't think it's redundant.
However, it is quite hard to read. I recommend adding a blank line after loading the `Your Datasets` page, and also between each upload. That along with the above comment about removing the assert statements should make it much more legible.